File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/Time.java
package aladdin.asn1.iso.pkix;
import aladdin.asn1.*; 

//	Time ::= CHOICE {
//		utcTime        UTCTime,
//		generalTime    GeneralizedTime 
//	}

public final class Time extends Choice
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(UTCTime		.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(GeneralizedTime.class).factory(), Cast.N), 
	}; 
	// конструктор
	public Time() { super(info); } 
}

File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/Time.cs
using System;

//	Time ::= CHOICE {
//		utcTime        UTCTime,
//		generalTime    GeneralizedTime 
//	}

namespace Aladdin.ASN1.ISO.PKIX
{
	public class Time : Choice
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<UTCTime		>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<GeneralizedTime>().Factory(), Cast.N), 
		}; 
		// конструктор
		public Time() : base(info) {} 
	}
}
